### PR TITLE
TST/DOC: stats: explain/check 100% interval for discrete distributions

### LINKED ISSUE #11026: rv_discrete.interval returning wrong values for alpha = 1
Scipy's rv_discrete.interval method, it appears that the lower bound is decreased by 1 when the alpha is set to exactly 1 (as far as I can tell, this is the most reliable way to get a generic rv's support's minimum)

#### Reproducing code example:

```
from scipy.stats import bernoulli  # also tested with randint and binom

rv = bernoulli(0.5)

print(rv.interval(1))  # expected (0,1), got (-1,1)
```

#### Scipy/Numpy/Python version information:

1.3.1 1.17.3 sys.version_info(major=3, minor=8, micro=0, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-11026

#### What does this implement/fix?
gh-11026 reported that discrete distributions returned incorrect values for `interval` with `confidence=1` (i.e. 100% confidence interval). They are not incorrect, per se, but it is somewhat surprising that the left end of the interval is beyond the support (as explained [here](https://github.com/scipy/scipy/issues/11026#issuecomment-1094200367)). This adds a note about this to the docs and adds a test to check for consistent behavior across discrete distributions.

### CODE REVIEW

**Comment by h-vetinari** [comment]
I commented on the issue, but I don't think this is actually very illuminating. If we decide we do not want to change this behaviour, then we should be more honest about the fact that this is scipy's implementation choice, i.e. I think the following falsely gives the impression that the `-1` somehow follows from the definition, but it doesn't (AFAICT).
```
        Suppose ``[c, d]`` is the support of a discrete distribution; then ``ppf([0, 1]) == (c-1, d)``
```

**Comment by mdhaber** [comment]
> I think the following falsely gives the impression that the -1 somehow follows from the definition, but it doesn't (AFAICT).

I meant `ppf` in the sense of SciPy's implementation, you're right. Boost does what we do while R `qbinom` at `0` returns 0.

So I think you're right that there was some flexibility in the definition (because for discrete distributions, the definition of `ppf` cannot simply be "the inverse of the `cdf`" _and_ be defined on all of `[0, 1]`). But here is one argument in favor of `ppf(0) = a - 1` : `cdf(ppf(0))` _does_ equal 0 and `ppf(cdf(a-1))` _does_ equal `a-1`. The same is not true if `ppf(0) = a`.

**Comment by josef-pkt** [comment]
I just got confused about ppf and interval again. Each time it takes me half an hour to work my way through it.
I convinced myself again that ppf and interval are correctly defined.

The best way for me is to look at a step function graph, that has a zero point at -1, and then read the step function as inverse function flipping axes.

**Comment by josef-pkt** [comment]
about the `interval`

My preferred terminology is now "equal-tailed" intervals.
Some statisticians and part of my statsmodels work uses "central"

It bins the probability in each tail, but does not try to find a shorter interval with the same required coverage.

I was trying to verify that the coverage of poisson.interval is correct, but got briefly confused that I also had the required coverage if I dropped the point at the lower bound.
(I'm adding poisson tolerance intervals to statsmodels and had to look at this recently.)

**Comment by mdhaber** [comment]
@josef-pkt thanks! Since you've thought through it again, do you have any suggestions for the PR, which aims to state what SciPy does and check that we do it consistently?

**Comment by josef-pkt** [comment]
> I convinced myself again that ppf and interval are correctly defined.

maybe I should amend this. 
I was surprised again by the -1 lower bound, even if it is consistent with the ppf definition.

Clipping at lower bound of distribution `.a` would make more sense for the definition of the `interval`, given that it is a closed interval that includes both boundary points.

I never really ran into this because the usual usage I have is for 0.9 or 0.95 coverage.
(If users want to have 100% coverage in a tolerance interval in my poisson function, then they would get out of bounds -1.)

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1532,6 +1532,15 @@ def interval(self, confidence=None, *args, **kwds):
             end-points of range that contain ``100 * alpha %`` of the rv's
             possible values.
 
+        Notes
+        -----
+        This is typically implemented as ``ppf([p_tail, 1-p_tail])``, where
```
Comment (josef-pkt) [inline scipy/stats/_distn_infrastructure.py:1537]: drop "typically"
it is implemented

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -1532,6 +1532,15 @@ def interval(self, confidence=None, *args, **kwds):
...
+        ``ppf`` is the inverse cumulative distribution function and
+        ``p_tail = (1-confidence)/2``. Suppose ``[c, d]`` is the support of a
+        discrete distribution; then ``ppf([0, 1]) == (c-1, d)``. Therefore,
+        when ``confidence==1`` and the distribution is discrete, the left end
+        of the interval will be beyond the support of the distribution.
```
Comment (josef-pkt) [inline scipy/stats/_distn_infrastructure.py:1542]: add something like

In discrete distributions the interval will limit the probability in each tail to to be less or equal to p_tail.

Comment (josef-pkt) [reply scipy/stats/_distn_infrastructure.py:1542]: and will in general be strictly less

File: scipy/stats/_distn_infrastructure.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/_distn_infrastructure.py:1537]: Good idea. I meant that a distribution could override the generic implementation, but I don't think any do.

**Review by tirthasheshpatel — APPROVED** [review]
The doc change by itself looks good. As far as this PR and gh-11026 are concerned, the discussion about changing the interval/support method seems out of scope (because I think it's just a convention SciPy has chosen rather than a bug so we can't change the method without breaking back-compatibility; the interval/support method change seems like a good candidate for gh-15928). I am happy to merge this later this week if there are no concerns.

**Comment by tirthasheshpatel** [comment]
Merged! Thanks @mdhaber @josef-pkt!

**Pull request merged**